inc/utils/namespace.php

```diff
@@ -1174,10 +850,21 @@ function clickhouse_query( string $query, string $body = '', string $return = 'a
 	$result = array_map( 'json_decode', $result );
 
 	// For single or zero results assume this is just a row of aggregate values and return it.
-	if ( $return === 'object' || ( $return === 'auto' && count( $result ) <= 1 ) ) {
+	if ( $return === 'object' ) {
 		return reset( $result );
 	}
 
 	// Return array.
 	return $result;
 }
+
+/**
+ * Get a unique cache key from a set of arguments.
+ *
+ * @param string $prefix The cache key prefix.
+ * @param mixed ...$args List of arguments to generate a cache key from.
+ * @return string
+ */
+function get_cache_key( string $prefix, ...$args ) : string {
+	return sprintf( '%s:%s', $prefix, hash( 'crc32', serialize( $args ) ) );
```

Comment:
Why crc32?

Reply:
crc32 is a data integrity hash designed to ensure that your data hasn't been corrupted in transit/etc, and to my knowledge is not specifically designed to produce an [avalanche effect](https://en.wikipedia.org/wiki/Avalanche_effect); in other words, the likelihood of collision is much higher.

Performance isn't relevant for this few cycles, and you're right that security isn't really a factor here (specifically: reversibility of the hash); collision-avoidance is the most desirable property. Unless there's a good reason not to, I'd use `sha1()` which has good collision properties and is the most "standard" hash.

Reply:
fair enough then, updating